graylog2-web-interface/src/components/indices/IndicesMaintenanceDropdown.jsx

```diff
@@ -9,22 +9,27 @@ import StoreProvider from 'injection/StoreProvider';
 const DeflectorStore = StoreProvider.getStore('Deflector'); // eslint-disable-line no-unused-vars
 
 const IndicesMaintenanceDropdown = React.createClass({
+  propTypes: {
+    indexSetId: React.PropTypes.string.isRequired,
+  },
+
   _onRecalculateIndexRange() {
     if (window.confirm('This will trigger a background system job. Go on?')) {
+      // TODO 2.2: Only rebuild index set here? It currently rebuilds all index ranges!
```

Comment:
For now I would simply change the wording to "all index ranges" and fix this post-beta.
Either by moving the action to the index sets overview page, or by implementing another system job that operates on a single index set.